pkg/hive/hive.go

```diff
@@ -187,6 +194,11 @@ func (s *Service) sendPeers(ctx context.Context, peer swarm.Address, peers []swa
 			return err
 		}
 
+		// Don't advertise private addresses to the public network.
+		if isPeerPublic && manet.IsPrivateAddr(addr.Underlay) {
```

Comment:
maybe it's better to have this check under `BroadcastPeers` so that you can send full batches instead of skipping some here.

Reply:
can we also have this check before we attempt to ping the address under `checkAndAddPeers`